consul-lambda-registrator/client/lambda.go

```diff
@@ -0,0 +1,211 @@
+package client
```

Comment:
I'm a bit hesitant about sharing this between the Lambda extension and Lambda registrator. It doesn't feel like there is a ton of overlap between use cases. What do you think?

Reply:
I do agree that there isn't much overlap between the two cases. What do you think about having different Lambda clients within this package? Or are you thinking that we should just put the clients in the registrator or extension, respectively?

Reply:
I'm thinking that this should probably be separated out into the extension and registrator.

Reply:
I've moved the Lambda client for the extension into the `extension` package and removed the functionality that the registrator uses.